lib/ClangImporter/ImportType.cpp

```diff
@@ -2724,8 +2724,10 @@ static ParamDecl *getParameterInfo(ClangImporter::Implementation *impl,
   // C++ types taking a reference might return a reference/pointer to a
   // subobject of the referenced storage. In those cases we need to prevent the
   // Swift compiler to pass in a temporary copy to prevent dangling.
-  if (ASTContext.LangOpts.hasFeature(Feature::AddressableParameters) &&
-      !param->getType().isNull() && param->getType()->isReferenceType()) {
+  if (!param->getType().isNull() && param->getType()->isReferenceType() &&
```

Comment:
Could we make this stricter and check for `param->getType()->isLValueReferenceType()`? Unfortunately, I don't think we can make addressable parameters compatible with rvalue references at the moment. The consume analysis relies on the SIL values being copied so once we eliminate those copies we no longer have the guarantees on the Swift side to never use moved-from objects.